python/dgl/transforms/functional.py

```diff
@@ -3856,4 +3857,133 @@ def double_radius_node_labeling(g, src, dst):
 
     return F.tensor(z, F.int64)
 
+def shortest_dist(g, root=None, directed=True, return_paths=False):
+    r"""Compute shortest distance and paths on the given graph.
+
+    This function computes the shortest distance between all pairs of nodes
+    or between all nodes and a specified root node. Corresponding shortest
+    paths can be returned and represented with edge IDs. Note that only
+    unweighted cases are allowed.
```

Comment:
I think you only need to mention that this function does not support weighted graphs. The rest can be covered by the descriptions of the parameters and the returned results.